### ISSUE TITLE
DynamoDB support for online feature store

### ISSUE THREAD

**Issue by blvp** [feast-dev/feast#1409]
**Is your feature request related to a problem? Please describe.**
We need to have an online serving backend with Dynamo to support customer deployments on AWS.

**Describe the solution you'd like**
Implementation of `Provider` similar to [/sdk/python/feast/infra/local_sqlite.py](https://github.com/feast-dev/feast/blob/master/sdk/python/feast/infra/local_sqlite.py) with DynamoDB backend

**Comment by woop** [comment]
Thanks @blvp.

As mentioned before, a complete implementation should contain
1. A provider (`aws`), which is used to update infrastructure during `feast apply`
2. Offline store support for this provider. Ideally Snowflake and/or Redshift
3. Online store support for this provider. Dynamo makes sense

To be clear, we can consider (2) to be out of scope for this card, but there may be some work required in creating the `aws` provider. Our plan is for all our providers (`local`, `gcp`) to be split from our online stores and offline stores. So the object through which the FeatureStore object can update infrastructure based on FeatureView/Table changes, and also the object the FeatureStore object uses to retrieve an OnlineStore or OfflineStore object.

Right now we still have some work in flight. Splitting the OnlineStore and Provider being one. We also want to do some code refactoring to make sure that the FeatureStore class has a clean abstraction between itself and the Provider.

I'd recommend the following steps.
1. Document the data model that we'd like to use for DynamoDB in this issue before starting development
2. Create an isolated implementation of the DynamoDB I/O. With the idea that changes in the Feast code base wont force you to rewrite the connector.
3. Once we are ready with the first release of Feast 0.10, we can integrate the connector. It remains to be seen how we will integrate the connector, because we'd want AWS users to have a complete experience (ideally with a production offline store). We may have to document this connector as `experimental` or `contrib` for the start.

**Comment by blvp** [comment]
Thank you @woop. Will update the ticket with the data model design.
